File: static/js/track.js
function Track(id) {
  this.id = id;
}

Track.prototype = {
  info_loaded: false,

  retrieveInfo: function(callback) {
    if(this.info_loaded == true) {
      callback();
    } else {
      var _this = this;
      $.get('/tracks/'+this.id+'.json', function(r) {
        _this.name = r.name;
        _this.artist_name = r.artist_name;
        _this.file_url = r.file_url;
        _this.genres = r.genres;
        _this.city = r.city;
        _this.state = r.state;
        _this.country = r.country;
        _this.info_loaded = true;
        callback();
      });
    }
  }
}

File: models/track.rb
class Track
  include DataMapper::Resource
  property :id, Serial
  property :jamendo_id, Integer, :index => true
  property :album_id, Integer
  property :genre_id, Integer, :index => true
  property :name, String, :length => 255
  property :file_name, String, :length => 255
  property :mbg_id, String, :length => 255
  property :num_album, String, :length => 255
  property :license, String, :length => 255
  belongs_to :album
  belongs_to :genre
  has n, :taggings
  has n, :tags, :through => :taggings

  def to_json(opts={})
    j = {'id' => id, 'name' => name.clean_up, 'artist_name' => album.artist.name.clean_up, 'file_url' => file_url}
    j['genres'] = {'id' => genre.id, 'name' => genre.name} if genre
    j['city'] = album.artist.city.name if album.artist.city
    j['subdivision'] = album.artist.subdivision.name if album.artist.subdivision
    j['country'] = album.artist.country.name if album.artist.country
    opts[:to_json] == false ? j : j.to_json
  end

  def file_url; "http://api.jamendo.com/get2/stream/track/redirect/?id=#{jamendo_id}&streamencoding=mp31" end
end